graylog2-web-interface/src/views/components/queries/QueryTitleEditModal.jsx

```diff
@@ -0,0 +1,91 @@
+// @flow strict
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Modal, Button } from 'components/graylog';
+import Input from 'components/bootstrap/Input';
+
+import type { TitlesMap } from 'views/stores/TitleTypes';
+
+/**
+ * Component that allows the user to update a QueryTab title.
+ * It takes the active query title as an argument on open and will use it as the draft.
+ * The open action is getting called outside by referencing this component.
+ * The state of this modal can't be handled inside the QueryTitle, due to the react bootstrap tabs key bindings.
+ * E.g. using the arrow keys inside the title input would result in losing the focus.
+ */
```

Comment:
I am not a big fan of that kind of comments, since they tend to be obsolete and out of date quite soon.

Reply:
I don't think we are going to remove the react bootstrap tabs soon, are we?
But I get your pint and will adjust it! Would you remove the full comment? I thought it is nice to have this main component comment for the UI frontend docs.

Reply:
I would suggest removing the last two lines.